components/mdns/mdns.c

```diff
@@ -4180,9 +4181,10 @@ void mdns_preset_if_handle_system_event(void *arg, esp_event_base_t event_base,
         default:
             break;
         }
-    }
+    } else
```

Comment:
I think this won't compile when `CONFIG_ESP_WIFI_ENABLED==1 && CONFIG_ETH_ENABLED==0`